### ISSUE TITLE
Incorrect error message when attempting to access field on struct constructor

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust#27990]
[playpen](http://is.gd/rb25Yu)

``` rust
struct MyStruct { field: u8 }

fn main() {
    MyStruct.foo;
}
```

results in

```
<anon>:4:5: 4:13 error: `MyStruct` is the name of a struct or struct variant, but this expression uses it like a function name [E0423]
<anon>:4     MyStruct.foo;
             ^~~~~~~~
<anon>:4:5: 4:13 help: see the detailed explanation for E0423
<anon>:4:5: 4:13 help: did you mean to write: `MyStruct { /* fields */ }`?
error: aborting due to previous error
```

in particular

> but this expression uses it like a function name

which doesn't make sense in this context. the same error message appears if one attempts `MyStruct()`, which makes more sense.

**Comment by steveklabnik** [comment]
Triage: new error format, but the underlying issue is still there:

```text
error[E0423]: `MyStruct` is the name of a struct or struct variant, but this expression uses it like a function name
 --> <anon>:4:5
  |
4 |     MyStruct.foo;
  |     ^^^^^^^^ struct called like a function
  |
  = help: did you mean to write: `MyStruct { /* fields */ }`?
```

**Comment by petrochenkov** [comment]
Error on non-ancient version of rustc:
```rust
rustc 1.17.0-nightly (be760566c 2017-02-28)
error[E0423]: expected value, found struct `MyStruct`
 --> <anon>:4:5
  |
4 |     MyStruct.foo;
  |     ^^^^^^^^ did you mean `MyStruct { /* fields */ }`?

error: aborting due to previous error
```

**Comment by petrochenkov** [comment]
For `MyStruct()`:
```rust
rustc 1.17.0-nightly (be760566c 2017-02-28)
error[E0423]: expected function, found struct `MyStruct`
 --> <anon>:4:5
  |
4 |     MyStruct();
  |     ^^^^^^^^ did you mean `MyStruct { /* fields */ }`?

error: aborting due to previous error
```

Seems fixed.
